pkg/services/meta/containers.go

```diff
@@ -591,3 +593,121 @@ func (s *containerStorage) handleNewEpoch(e uint64) error {
 
 	return errors.Join(mptErr, dbErr)
 }
+
+func expandChildren(st storage.Store, rootOIDs []byte) []byte {
+	if len(rootOIDs) == 0 {
+		return nil
+	}
+
+	// sorting before every SEEK is used for exact single operation for every
+	// children searching subroutine; search is done in 3 steps:
+	//  1. root -> last/link object search
+	//  2. last/link -> first part ID search
+	//  3. first part ID -> all children search
+
+	var childrenWithParentID [][]byte
+	var rng storage.SeekRange
+	rng.Prefix = slices.Concat([]byte{attrPlainToOIDIndex}, []byte(objectsdk.FilterParentID), object.AttributeDelimiter)
+	rng.Start = make([]byte, oid.Size)
+
+	rootsSorted := slices.SortedFunc(slices.Chunk(rootOIDs, oid.Size), bytes.Compare)
+	lastRoot := rootsSorted[len(rootsSorted)-1]
+	copy(rng.Start, rootsSorted[0])
+
+	st.Seek(rng, func(k, _ []byte) bool {
+		currRoot := k[len(rng.Prefix) : len(rng.Prefix)+oid.Size]
+		if len(rootsSorted) == 0 || bytes.Compare(currRoot, lastRoot) > 0 {
```

Comment:
if we pre-cmp against `lastRoot` anyway, we can also catch equality case

Reply:
dropped this check at all

Reply:
it wasn't bad actually, but had micro effect, so be it